File: jenkins-shared-library/examples/Jenkinsfile.go
/**
 * Go application Jenkinsfile example
 */

@Library('fawkes-pipeline-library') _

goldenPathPipeline {
    appName = 'go-service'
    language = 'go'
    
    // Custom Go commands
    buildCommand = 'go build -v ./...'
    testCommand = 'go test -v -coverprofile=coverage.out ./...'
    bddTestCommand = 'go test -v ./features/...'
    
    notifyChannel = 'go-team'
}

File: jenkins-shared-library/examples/Jenkinsfile.java
/**
 * Java application Jenkinsfile example
 */

@Library('fawkes-pipeline-library') _

goldenPathPipeline {
    appName = 'java-service'
    language = 'java'

    // Custom Maven commands
    buildCommand = 'mvn clean package -DskipTests'
    testCommand = 'mvn test'
    bddTestCommand = 'mvn verify -Pcucumber'

    notifyChannel = 'java-team'
}
